src/vs/workbench/contrib/terminal/browser/xterm/decorationStyles.ts

```diff
@@ -24,6 +32,69 @@ export const enum DecorationSelector {
 	LightBulb = 'codicon-light-bulb'
 }
 
+export class TerminalDecorationHoverService extends Disposable {
+	private _hoverDelayer: Delayer<void>;
+	private _contextMenuVisible: boolean = false;
+
+	constructor(@IHoverService private readonly _hoverService: IHoverService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IContextMenuService contextMenuService: IContextMenuService) {
+		super();
+		this._register(contextMenuService.onDidShowContextMenu(() => this._contextMenuVisible = true));
+		this._register(contextMenuService.onDidHideContextMenu(() => this._contextMenuVisible = false));
+		this._hoverDelayer = this._register(new Delayer(configurationService.getValue('workbench.hover.delay')));
+	}
+
+	public hideHover(hoverDelayer?: Delayer<void>, hoverService?: IHoverService) {
+		if (hoverDelayer && hoverService) {
+			hoverDelayer.cancel();
+			hoverService.hideHover();
+			return;
+		}
+		this._hoverDelayer.cancel();
+		this._hoverService.hideHover();
+	}
+
+	createHover(element: HTMLElement, command: ITerminalCommand | undefined, hoverMessage?: string): IDisposable[] {
+		return [
```

Comment:
This will let you return a single `IDisposable`, which lets you remove the for loop to register it:

```ts
return combinedDisposable([...]);
```